libraries/core_libs/network/src/tarcap/packets_handlers/transaction_packet_handler.cpp

```diff
@@ -126,33 +141,40 @@ void TransactionPacketHandler::periodicSendTransactions(SharedTransactions &&tra
           continue;
         }
 
-        transactions_to_send[peer.first].push_back(trx->rlp());
-        transactions_hash_to_send[peer.first].push_back(trx_hash);
+        transactions_to_send[peer.first].first.push_back(trx_hash);
+        transactions_to_send[peer.first].second.push_back(trx->rlp());
       }
     }
   }
 
   LOG(log_tr_) << "Sending Transactions " << transactions_to_log << " to " << peers_to_log;
 
   for (auto &it : transactions_to_send) {
-    sendTransactions(it.first, it.second);
+    sendTransactions(it.first, it.second.first, it.second.second);
   }
-  for (auto &it : transactions_hash_to_send) {
-    for (auto &it2 : it.second) {
+  for (auto &it : transactions_to_send) {
+    for (auto &it2 : it.second.first) {
       peers[it.first]->markTransactionAsKnown(it2);
```

Comment:
this marking should be inside sendTransactions function and only if sealAndSend returned true